### ISSUE TITLE
<script> templates should not be processed by template7Page option

### ISSUE THREAD

**Issue by m-emelchenkov** [nolimits4web/Framework7#1066]
I've attached the code which illustrates the issue. Try to comment "template7Pages: true," and compare results.

[f7_bugreport.zip](https://github.com/nolimits4web/Framework7/files/365643/f7_bugreport.zip)

**Comment by m-emelchenkov** [comment]
OK, I dig into sources and find out that it seems it is not a critical bug. Because the framework compiles <script> templates before processing whole page, if I understand right. Anyway, it should not touch <script> portions inside resulting html, but at least I have access to compiled versions via  Template7.templates variable.

But: Template7.templates.myTemplate works in previously attached sample code, but does not work in my project (it is "undefined"). Any thoughts why?

**Comment by m-emelchenkov** [comment]
I found that it happens when I navigate via 'mainView.router.loadPage()' from index page to another page, and trying to use template in another page (the same as in index, in 'onPageInit'). Then Template7.templates object still exists but there are no templates in it.

**Comment by m-emelchenkov** [comment]
One more thought: app.initTemplate7Templates() called only once at app init, so no \<script\> templates can be used in another pages, I mean I need to put all \<script\> templates in index? Can you please add this approach in documentation? Only if it can't be fixed, of course. It's much more natural for me to store templates alongside with page layout. At least, I solve my problem when everything were broken :)

**Comment by ZanderBrown** [comment]
I'm not sure what your issue is.

While I admit it would be nice to have templates automatically done per page the performance would drop wouldn't be great and the Template7.compile() method can be called on demand.

**Comment by m-emelchenkov** [comment]
> While I admit it would be nice to have templates automatically done per page the performance would drop wouldn't be great and the Template7.compile() method can be called on demand.

Yes, this. And while it is not implemented just improve docs. I spend some time trying to understand how working with templates should be done.

**Comment by valnub** [comment]
I guess this issue is solved
